app/models/mixins/relationship_mixin.rb

```diff
@@ -150,21 +163,32 @@ def parent_count(*args)
   def parent_rel(*args)
     rels = parent_rels(*args).take(2)
     raise _("Multiple parents found.") if rels.length > 1
+
     rels.first
   end
 
   # Returns the parent of the record, nil for a root node
   def parent(*args)
-    rels = parent_rels(*args).take(2)
-    raise _("Multiple parents found.") if rels.length > 1
-    rels.first.try(:resource)
+    if @relationship_type == :genealogy
+      call_ancestry_method(:parent, binding)
+    else
+      rels = parent_rels(*args).take(2)
+      raise _("Multiple parents found.") if rels.length > 1
+
+      rels.first.try(:resource)
+    end
   end
 
   # Returns the class/id pair of the parent of the record, nil for a root node
   def parent_id(*args)
-    rels = parent_ids(*args).take(2)
-    raise _("Multiple parents found.") if rels.length > 1
-    rels.first
+    if @relationship_type == :genealogy
+      call_ancestry_method(:parent_id, binding)
+    else
+      rels = parent_ids(*args).take(2)
```

Comment:
REALLY don't like this

this is here because we often end up with too many references.
moving to `belongs_to` (phase 2) will be great to remove these

you didn't do this I know but... I still don't like it